tests/src/JIT/SIMD/Vector3Interop.cs

```diff
@@ -0,0 +1,339 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Numerics;
+using System.Runtime.InteropServices;
+using System.Text;
+
+
+public struct DT 
+{
+    public Vector3 a;
+    public Vector3 b;
+};
+
+[StructLayout(LayoutKind.Sequential, CharSet = CharSet.Ansi)]
+public struct ComplexDT 
+{
+    public int iv;
+    public DT vecs;
+    [MarshalAs(UnmanagedType.ByValTStr, SizeConst=256)] public string str;
+    public Vector3 v3;
+};
+
+class PInvokeTest 
+{
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern int nativeCall_PInvoke_CheckVector3Size();
+
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern float nativeCall_PInvoke_Vector3Arg(
+        int i, 
+        Vector3 v1, 
+        [MarshalAs(UnmanagedType.LPStr)] string s, 
+        Vector3 v2);
+
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern Vector3 nativeCall_PInvoke_Vector3Ret();
+
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern float nativeCall_PInvoke_Vector3Array(Vector3[] v_array);
+
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern DT nativeCall_PInvoke_Vector3InStruct(DT d);
+
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern void nativeCall_PInvoke_Vector3InComplexStruct(ref ComplexDT cdt);
+    
+    public static bool test() 
+    {
+
+        // Expected return value is 12 bytes.
+        if (nativeCall_PInvoke_CheckVector3Size() != 12) 
```

Comment:
If native Vector3 size is 12 bytes,  a couple of questions here

1) Is native compiler rounding the size of native Vector3 struct to 16 bytes like x64 CLR does?
2) While passing a native vector3 struct as arg, just 12 bytes passed or 16 bytes?
If it is just 12 bytes, in RPInvoke scenario is RyuJIT reading past the end of struct?  If it is 16 bytes, is native compiler ensuring that the upper 4 bytes of the struct are zero initialized?  Specifically does native compiler spec (on x64 windows and unix) mention that it would zero either round the size to 16 bytes while passing as arg or zero init upper 4 bytes to zero?